storage/storage.go

```diff
@@ -0,0 +1,63 @@
+package storage
+
+import (
+	"fmt"
+	"os"
+	"sync"
+)
+
+const k6BrowserDataDirPattern = "xk6-browser-data-*"
+
+// Dir manages data storage for the extension and user specific data
+// on the local filesystem.
+type Dir struct {
+	Dir    string // path to the data storage directory
+	remove bool   // whether to remove the temporary directory in cleanup
+
+	// FS abstractions
+	fsMkdirTemp       func(dir, pattern string) (string, error)
+	fsRemoveAll       func(path string) error
+	fsRemovalAllMutex sync.Mutex
```

Comment:
You mean `fsRemoveAllMutex`—`Removal`? It's a little style thing, but you can shorten `Mutex` to `Mu`. It's a widely accepted convention, so people can see what's it about.